src/main/java/bio/terra/service/DatasetService.java

```diff
@@ -0,0 +1,291 @@
+        FlightMap flightMap = new FlightMap();
+        flightMap.put(JobMapKeys.DESCRIPTION.getKeyName(), "Create dataset " + datasetRequestModel.getName());
+        flightMap.put("request", datasetRequestModel);
+        // TODO: wire this up
+        //return stairway.submit(DatasetCreateFlight.class, flightMap);
+        throw new ApiException("Create dataset is not implemented");
+    }
+
+    /**
+     * Kick-off dataset deletion
+     *
+     * @param id dataset id to delete
+     * @returns jobId (flightId) of the job
+     */
+    public String deleteDataset(String id) {
+        FlightMap flightMap = new FlightMap();
+        flightMap.put(JobMapKeys.DESCRIPTION.getKeyName(), "Delete dataset " + id);
+        flightMap.put("id", id);
+        // TODO: wire this up
+        //return stairway.submit(DatasetDeleteFlight.class, flightMap);
+        throw new ApiException("Delete dataset is not implemented");
+    }
+
+    /**
+     * Enumerate a range of datasets ordered by created date for consistent offset processing
+     * @param offset
+     * @param limit
+     * @return list of summary models of dataset
+     */
+    public List<DatasetSummaryModel> enumerateDatasets(int offset, int limit) {
+        return datasetDao.retrieveDatasets(offset, limit)
+                .stream()
+                .map(summary -> makeSummaryModelFromSummary(summary))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Return a single dataset summary given the dataset id.
+     * This is used in the create dataset flight to build the model response
+     * of the asynchronous job.
+     *
+     * @param id
+     * @return summary model of the dataset
+     */
+    public DatasetSummaryModel retrieveDatasetSummary(UUID id) {
+        DatasetSummary datasetSummary = datasetDao.retrieveDatasetSummary(id);
+        return makeSummaryModelFromSummary(datasetSummary);
+    }
+
+    /**
+     * Return the output form of dataset
+     * @param id
+     * @return dataset model
+     */
+    public DatasetModel retrieveDataset(UUID id) {
+        Dataset dataset = datasetDao.retrieveDataset(id);
+        return makeDatasetModelFromDataset(dataset);
+    }
+
+    /**
+     * Make a Dataset structure with all of its parts from an incoming dataset request.
+     * Note that the structure does not have UUIDs or created dates filled in. Those are
+     * updated by the DAO when it stores the dataset in the repository metadata.
+     *
+     * @param datasetRequestModel
+     * @return Dataset
+     */
+    public Dataset makeDatasetFromDatasetRequest(DatasetRequestModel datasetRequestModel) {
+        // Make this early so we can hook up back links to it
+        Dataset dataset = new Dataset();
+
+        List<DatasetRequestContentsModel> requestContentsList = datasetRequestModel.getContents();
+        // TODO: for MVM we only allow one source list
+        if (requestContentsList.size() > 1) {
```

Comment:
Out of the scope of this PR, but after we get this and Rori's validation PR in I think we should move this (and the TODO) to the `DatasetRequestValidator` to keep structural validation in the same place.

Reply:
I think we need a comment here, because the .get(0) looks pretty suspect unless you know that is a temporary restriction. But happy to have the validation do the check.